src/aspire/utils/preprocess.py

```diff
@@ -139,20 +138,26 @@ def downsample(insamples, szout, mask=None):
         # stack of one dimension objects
 
         for idata in range(ndata):
-            insamples_fft = crop_pad(fftshift(fft(insamples[idata])), L_out)*mask
-            outsamples[idata] = np.real(ifft(ifftshift(insamples_fft))*(L_out / L_in))
+            insamples_fft = crop_pad(xp.asnumpy(fft.fftshift(
```

Comment:
Some of these one-liners become quite hard to parse when we add the `asarray` and `asnumpy` functions. Maybe split up into several lines?

Reply:
Changed to multiple lines in commit ff09206.

Reply:
They're still single statements, just with many linebreaks in them. What I meant is split it up into several statements and assign to intermediate variables.

Reply:
Changed in commit 5164992.